firebase-latest-version-update-checker.js

```diff
@@ -0,0 +1,39 @@
+import AbstractUpdateChecker from './abstract-update-checker';
+export default class FirebaseLatestVersionUpdateChecker extends AbstractUpdateChecker {
+
+  constructor({ fbDatabase, latestVersionPath, curVersion }) {
+    super();
+
+    if (!fbDatabase || !latestVersionPath || !curVersion) {
+      throw new Error('Required config not found. Make sure you have specified' +
+        '`fbDatabase`, `latestVersionPath` and `curVersion`.')
+    }
+
+    this.fbDatabase = fbDatabase;
+    this.latestVersionPath = latestVersionPath;
+    this.curVersion = curVersion;
+    this._watchReleases();
+  }
+
+  async _buildQuery() {
+    return this.fbDatabase.ref(this.latestVersionPath);
+  }
+
+  /**
+   * It watches the `latestVersionPath`, and whenever a new release is found, sets `updates` property.
+   */
+  async _watchReleases() {
+    let query = await this._buildQuery();
+    query.on('value', (snapshot) => {
+      const latestVersion = snapshot.val();
+      this.updates = latestVersion === this.curVersion ? null : latestVersion;
```

Comment:
@nirmalbaldaniya-dream IMO, हमें यहा semver की library का use करके latest version आया हे या नहीं वो check करना चाहिए, जेसे हमने हमारी service-worker https://git.dream-world.in/dw/dw-pwa-helpers/-/blob/master/service-worker-manager.js#L177 में किया था.

Reply:
To avoid a dependency, we will continue without SemVer for now.